src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -3006,8 +3006,7 @@ private AssemblyTableInfo[] GetInstalledAssemblyTableInfo(bool ignoreInstalledAs
                     {
                         // Exactly one TargetFrameworkDirectory, so assume it's related to this
                         // InstalledAssemblyTable.
-
-                        frameworkDirectory = TargetFrameworkDirectories[0];
+                        frameworkDirectory = FileUtilities.EnsureTrailingSlash(TargetFrameworkDirectories[0]);
```

Comment:
Would this be appropriate? It "feels" like it expresses intention more clearly.

```suggestion
                        frameworkDirectory = FileUtilities.NormalizeForPathComparison(TargetFrameworkDirectories[0]);
```

Reply:
I think that change is reasonable, but if we take it, we'd have to change all the other points that EnsureTrailingSlash rather than NormalizeForPathComparison, or else we'd be ensuring cache misses rather than increasing the chance of cache hits. (I can do that if you'd like, but it's a larger change, and it'd be easier to accidentally miss a case.)